datasets/wikicorpus/wikicorpus.py

```diff
@@ -0,0 +1,179 @@
+"""
+
+_HOMEPAGE = "https://www.cs.upc.edu/~nlp/wikicorpus/"
+
+_LICENSE = "GNU Free Documentation License"
+
+_URLs = "https://www.cs.upc.edu/~nlp/wikicorpus/{form}.{language}.tgz"
+
+_LANGUAGES = ["ca", "es", "en"]
+_FORMS = ["raw", "tagged"]
+
+METADATA_PATTERN = re.compile(r'.+id="(?P<id>[^"]+)".+title="(?P<title>[^"]+)".+')
+
+
+class WikicorpusConfig(datasets.BuilderConfig):
+    """ BuilderConfig for Wikicorpus."""
+
+    def __init__(self, form=None, language=None, **kwargs):
+        """
+        Args:
+            form: form of the dataset.
+            language: language of the dataset.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super().__init__(
+            name=f"{form}_{language}",
+            description=f"Wikicorpus dataset in {form} form and {language} language.",
+            **kwargs,
+        )
+        self.form = form
+        self.language = language
+
+
+class Wikicorpus(datasets.GeneratorBasedBuilder):
+    """Wikicorpus dataset."""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIG_CLASS = WikicorpusConfig
+    BUILDER_CONFIGS = [WikicorpusConfig(form=form, language=language) for form in _FORMS for language in _LANGUAGES]
+
+    def _info(self):
+        if self.config.form == "raw":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.form == "tagged":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "sentence": datasets.Sequence(datasets.Value("string")),
+                    "lemmas": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.Value("string")),
+                    "wordnet_senses": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        url_to_download = _URLs.format(form=self.config.form, language=self.config.language)
+        downloaded_dir = dl_manager.download_and_extract(url_to_download)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "dirpath": downloaded_dir,
+                },
+            ),
+        ]
+
+    def _generate_examples(self, dirpath):
+        for filepath in Path(dirpath).iterdir():
```

Comment:
```suggestion
        for filepath in sorted(Path(dirpath).iterdir()):
```
we need to make this deterministic